quests/rl/trainer/trainer.py

```
@@ -0,0 +1,188 @@
+    parser.add_argument(
+        '--explore_decay',
+        help='The rate at which to decay the probability of a random action',
+        type=float,
+        default=0.01)
+    parser.add_argument(
+        '--memory_size',
+        help='Size of the memory buffer',
+        type=int,
+        default=100000)
+    parser.add_argument(
+        '--memory_batch_size',
+        help='The amount of memories to sample from the buffer while training',
+        type=int,
+        default=64)
+    parser.add_argument(
+        '--training_mode',
+        help='True for the model to train, False for the model to evaluate',
+        type=bool,
+        default=True)
+    parser.add_argument(
+        '--model_path',
+        help='Directory where to save the given model',
+        type=str,
+        default='models/')
+    parser.add_argument(
+        '--save_model',
+        help='Whether to save the model',
+        type=bool,
+        default=True)
+    parser.add_argument(
+        '--score_print_rate',
+        help='How often to print the score, 0 if never',
+        type=int,
+        default=0)
+    parser.add_argument(
+        '--render_rate',
+        help='How often to render an episode, 0 if never',
+        type=int,
+        default=0)
+    parser.add_argument(
+        '--record_rate',
+        help="""How often to record metrics to tensorboard, 0 if never, higher
+        to avoid hyperparameter tuning "too many metrics" error""",
+        type=int,
+        default=100)
+    return parser.parse_known_args(argv)
+
+
+def _run(args):
+    """Sets up and runs the gaming simulation.
+
+    Initializes TensorFlow, the training agent, and the game environment.
+    The agent plays the game from the starting state for a number of
+    episodes set by the user.
+
+    Args:
+      args (dict): The arguments from the command line parsed by
+        _parse_arguments.
+    """
+    with tf.Session() as sess:
+        env = gym.make(args.game)
```

maybe we can raise an exception if game doesnt exist?

---

Could you help me understand your suggestion better? gym already throws an error if the environment does not exist:
gym.error.UnregisteredEnv: No registered env with id: LunarLanders-v2

---

Yeah,I was thinking of maybe something more descriptive for the Qwiklab audience but this should be fine